haystack/preview/components/websearch/serper_dev.py

```diff
@@ -0,0 +1,133 @@
+        )
+        headers = {"X-API-KEY": self.api_key, "Content-Type": "application/json"}
+
+        try:
+            response = requests.post(url, headers=headers, data=payload, timeout=30)
+            response.raise_for_status()  # Will raise an HTTPError for bad responses
+        except requests.Timeout:
+            raise TimeoutError(f"Request to {self.__class__.__name__} timed out.")
+
+        except requests.RequestException as e:
+            raise Exception(f"An error occurred while querying {self.__class__.__name__}. Error: {e}")
+
+        # If we reached this point, it means the request was successful and we can proceed
+        json_result = response.json()
+
+        # we get the snippet from the json result and put it in the content field of the document
+        organic = [
+            Document.from_dict({"metadata": {k: v for k, v in d.items() if k != "snippet"}, "content": d["snippet"]})
+            for d in json_result["organic"]
+        ]
+
+        answer_box = []
+        # answer box is what search engine shows as a direct answer to the query
+        if "answerBox" in json_result:
+            answer_dict = json_result["answerBox"]
+            highlighted_answers = answer_dict.get("snippetHighlighted")
+            answer_box_content = None
+            # Check if highlighted_answers is a list and has at least one element
+            if isinstance(highlighted_answers, list) and len(highlighted_answers) > 0:
+                answer_box_content = highlighted_answers[0]
+            elif isinstance(highlighted_answers, str):
+                answer_box_content = highlighted_answers
+            if not answer_box_content:
+                for key in ["snippet", "answer", "title"]:
+                    if key in answer_dict:
+                        answer_box_content = answer_dict[key]
+                        break
+            if answer_box_content:
+                answer_box = [
+                    Document(
+                        content=answer_box_content,
+                        metadata={"title": answer_dict.get("title", ""), "link": answer_dict.get("link", "")},
+                    )
+                ]
+
+        people_also_ask = []
+        # these are related questions that search engine shows
+        if "peopleAlsoAsk" in json_result:
+            for result in json_result["peopleAlsoAsk"]:
+                title = result.get("title", "")
+                people_also_ask.append(
+                    Document(
+                        content=result["snippet"] if result.get("snippet") else title,
+                        metadata={"title": title, "link": result.get("link", None)},
+                    )
+                )
+
+        documents = answer_box + organic + people_also_ask
+
+        logger.debug("Serper Dev returned %s documents for the query '%s'", len(documents), query)
+        return documents[: self.top_k]
```

Comment:
@ZanSara do I need to return a dictionary with documents key?

Reply:
Yes :+1: